src/main/scala/com/courseWork/auth/User.scala

```diff
@@ -0,0 +1,17 @@
+package com.courseWork.auth
+
+import io.circe.Codec
+import io.circe.generic.semiauto._
+
+import java.util.UUID
+
+case class User(
+    uid: UUID,
+    nickname: String,
+    email: String,
+    password: String
```

Comment:
Make type validation here on Model parsing, do not allow external systems provide wrong values to your logic. It will allow you to simplify the logic of `AuthService` by removing at least `ValidatedNec` there